lib/DiscoveryStreamFeed.jsm

```diff
@@ -181,6 +193,16 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
         Cu.reportError("No response for response.layout prop");
       }
     }
+    return layout;
+  }
+
+  async loadLayout(sendUpdate, isStartup) {
+    let layout = {};
+    if (this.config.hardcoded_layout) {
+      layout = {lastUpdate: Date.now(), ...defaultLayoutResp};
```

Comment:
This bypasses the call to `fetchLayout` which adds a layout entry to the cache. Currently the cache is used to determine if things should be expired, so lacking the entry makes it seem like the layout is always expired:
https://github.com/mozilla/activity-stream/blob/48d1f4c662c7f086ee9691272c9a0e79203b3c0d/lib/DiscoveryStreamFeed.jsm#L157-L161

And because layout appears to be expired, it triggers `loadLayout` leading to the spocs endpoint being "updated" and sets loaded status to false and prevents DiscoveryStreamBase from rendering.